tests/test.py

```diff
@@ -33,28 +47,173 @@ def test_multi_line_condition():
 
 
 # BAD (use parentheses in both case of line continuation)
 def test_parentheses_in_if_only_with_first_new_line():
     s = """if (
-    a == b): c + d
+a == b): c + d
     """
     assert _results(s)
 
 
+# GOOD (have all brackets on the same line)
+def test_list_in_one_line():
+    s = """a = [1, 2, 3]"""
+    assert not _results(s)
+
+
+# GOOD
+# (when the opening bracket is the last thing on a line, the matching closing
+# bracket is the first thing on a new line that matches the indentation of the
+# line with the opening bracket)
+def test_list_with_enters_line():
+    s = """a = [
+1, 2, 3
+]"""
+    assert not _results(s)
+
+
+# BAD (opening bracket is last, but closing is not on new line)
+def test_list_with_only_one_enter_line():
+    s = """a = [
+1, 2, 3]"""
+    assert _results(s)
+
+
+# BAD
+# (opening bracket is last, closing is on new line but indentation mismatch)
+def test_list_mismatch_line():
+    s = """a = [
+1, 2, 3
+    ]"""
+    assert _results(s)
+
+
+# GOOD (opening bracket is not last, so we don't care about the closing one)
+def test_list_open_bracket_not_last():
+    s = """a = [1, 2, 3
+]"""
+    assert not _results(s)
+
+
+# GOOD (opening bracket is not last, so we don't care about the closing one)
+def test_list_open_bracket_not_last_2():
+    s = """a = [1, 2, 3
+]"""
+    assert not _results(s)
+
+
+# (pretty much the same rules apply for multiple brackets)
+
+# GOOD (have all brackets on the same line)
+def test_multilist():
+    s = """a = [[1, 2, 3], [4, 5, 6]]"""
+    assert not _results(s)
+
+
+# GOOD
+def test_multilist_with_enters():
+    s = """a = [
+[1, 2, 3], [4, 5, 6]
+]"""
+    assert not _results(s)
+
+
+# GOOD
+def test_multilist_with_enters_2():
+    s = """a = [
+[1, 2, 3],
+[4, 5, 6]
+]"""
+    assert not _results(s)
+
+
+# GOOD
+def test_multilist_with_enters_3():
+    s = """a = [[
+1, 2, 3
+]]"""
+    assert not _results(s)
+
+
+# GOOD
+def test_multilist_with_enters_3():
+    s = """a = [
+    [
+        1, 2, 3
+    ],
+    [
+        4, 5, 6
+    ]
+]"""
+    assert not _results(s)
+
+
+# BAD
+# (opening bracket is last, closing is on new line but indentation mismatch)
+def test_multilist_mismatch():
+    s = """a = [
+    [
+        1, 2, 3
+    ],
+    [
+        4, 5, 6
+    ]
+  ]"""
+    assert _results(s)
+
+
+# BAD
+# (opening bracket is last, closing is on new line but indentation mismatch)
+def test_multilist_mismatch_2():
+    s = """a = [
+    [
+        1, 2, 3
+      ],
+    [
+        4, 5, 6
+    ]
+]"""
+    assert _results(s)
+
+
+# BAD
+# (opening bracket is last, closing is on new line but indentation mismatch)
+def test_multilist_mismatch_3():
+    s = """a = [
+    [
+        1, 2, 3
+    ],
+    [
+            4, 5, 6
+     ]
+]"""
+    assert _results(s)
+
+
+# BAD (if you have a closing bracket on a new line, don't open a new bracket)
+def test_multilist_mismatch_4():
+    s = """a = [[
+        1, 2, 3
+    ], [
+        4, 5, 6
+    ]]"""
```

Comment:
```suggestion
    s = """a = [[
    1, 2, 3
], [
    4, 5, 6
]]"""
```
The way the test was, it should actually produce two errors: in line 3 and in 5. Now it should only cause a error in line 5.

But that gives me ideas for more tests:

```python
# BAD 
# (1. if you have a closing bracket on a new line, don't open a new bracket
#  2. the first closing bracket's indentation is a missmatch) 
a = [[
    1, 2, 3
    ], [
    4, 5, 6
]]
```

```python
# BAD 
# (all opening brackets that are consecutive on the same line are on the same
# line should also have all closing brackets on the same line)
a = [[
    1, 2, 3
]
]
```